Lib/asyncio/streams.py

```diff
@@ -366,13 +372,12 @@ def get_extra_info(self, name, default=None):
             if exc is not None:
                 raise exc
         if self._transport.is_closing():
-            # Yield to the event loop so connection_lost() may be
-            # called.  Without this, _drain_helper() would return
-            # immediately, and code that calls
-            #     write(...); await drain()
-            # in a loop would never call connection_lost(), so it
-            # would not see an error when the socket is closed.
-            await sleep(0, loop=self._loop)
+            # Wait for protocol.connection_lost() call
+            # Raise connection closing error if any,
+            # ConnectionResetError otherwise
+            fut = self._protocol._get_close_waiter(self)
+            await fut
+            raise ConnectionResetError('Connection lost')
```

Comment:
Do we want to raise an error from drain()?  Maybe it should just return?  What's the point of this error?

Reply:
That's how `protocol._drain_helper()` works now.
The future is set to exception only if the connection is closed with a failure.
But `await writer.drain()` raises `ConnectionResetError`. 
I believe this is good: `await writer.write(b'data')` should fail loudly if the socket is closed.

Reply:
Maybe the code could be simplified by setting ConnectionResetError by `connection_lost()` callback handler but I'd like to keep the PR small.
Future improvement worth another pull request.

Reply:
OK, got it.